snuba/subscriptions/consumer.py

```diff
@@ -95,6 +95,9 @@ def revocation_callback(partitions: Sequence[Partition]) -> None:
             topics, on_assign=on_assign, on_revoke=revocation_callback
         )
 
+    def unsubscribe(self) -> None:
```

Comment:
I noticed we are not using it in this PR. Does the PR actually depends on it ?

Reply:
This is used in the `StreamTestMixin`, but not anywhere on production paths. It was previously defined on the Kafka consumer (which was the only concrete implementation tested by the mixin), so the tests would pass but not type check. It mostly just seemed easy enough to add it to the implementations across the board and leave the test as-is rather than go the other way.